Firestore/core/src/firebase/firestore/api/listener_registration.mm

```diff
@@ -16,14 +16,14 @@
 
 #include "Firestore/core/src/firebase/firestore/api/listener_registration.h"
 
-#import "Firestore/Source/Core/FSTFirestoreClient.h"
+#include "Firestore/core/src/firebase/firestore/core/firestore_client.h"
 
 namespace firebase {
 namespace firestore {
 namespace api {
 
 ListenerRegistration::ListenerRegistration(
-    FSTFirestoreClient* client,
+    std::shared_ptr<core::FirestoreClient> client,
     std::shared_ptr<core::AsyncEventListener<core::ViewSnapshot>>
         async_listener,
     std::shared_ptr<core::QueryListener> query_listener)
```

Comment:
Nit: `std::move` the argument into the member variable. In general, when a function takes non-primitive arguments by value and wishes to store them, it should move them to avoid an unnecessary copy.

Reply:
Done.